cpp/src/cylon/table.cpp

```diff
@@ -166,17 +159,44 @@ Status Table::Select(const std::function<bool(cylon::Row)> &selector, shared_ptr
   }
   return status;
 }
-Status Table::DistributedUnion(const shared_ptr<Table> &right, shared_ptr<Table> &out) {
+
+Status Table::DoSetOperation(SetOperation operation, const shared_ptr<Table> &right, shared_ptr<Table> &out) {
   std::string uuid = cylon::util::uuid::generate_uuid_v4();
-  cylon::Status status = cylon::DistributedUnion(ctx, this->id_, right->id_, uuid);
+  cylon::Status status = operation(this->ctx, this->id_, right->id_, uuid);
```

Comment:
Where is this operation() function located?

Reply:
its a typedef for a function pointer 
https://github.com/cylondata/cylon/blob/d62e8634b462ba8a6d78fa1b59a3a08956562ba3/cpp/src/cylon/table.hpp#L198

Reply:
Okay. Took some time to understand that. :)

What's the advantage of doing this? Even if directly call the source function here,

```c++
Status Table::Union(const shared_ptr<Table> &right, std::shared_ptr<Table> &out) {
  return DoSetOperation(&cylon::Union, right, out);
}
```

it will take just one line, right? Does this improve readability or maintainability?

Reply:
Yes, because we can reuse DoSetOperation method for Union, Subtract, Intersection, and their dist variants. Otherwise, we have to repeat the same piece of code in all of these places. 
https://github.com/cylondata/cylon/blob/d62e8634b462ba8a6d78fa1b59a3a08956562ba3/cpp/src/cylon/table.cpp#L172
Take a look at likes 172-194

Reply:
Got it. This is nice! Thanks.